scripts/verify_conda_libmesh.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+# * This file is part of the MOOSE framework
+# * https://www.mooseframework.org
+# *
+# * All rights reserved, see COPYRIGHT for full restrictions
+# * https://github.com/idaholab/moose/blob/master/COPYRIGHT
+# *
+# * Licensed under LGPL 2.1, please see LICENSE for details
+# * https://www.gnu.org/licenses/lgpl-2.1.html
+
+import sys
+import json
+from os import getenv
+from os.path import exists, join
+from glob import glob
+try:
+    from conda_build.metadata import MetaData
+    from conda.cli.python_api import Commands, run_command
+except ImportError:
+    sys.exit(0)
+
+MOOSE_DIR = getenv("MOOSE_DIR")
+CONDA_PREFIX = getenv("CONDA_PREFIX")
+
+
+def parse_recipe_meta_data(meta_file_path):
+    """Construct package version from metadata."""
+    meta = MetaData(meta_file_path)
+    pkg_name = meta.meta['package']['name']
+    pkg_version = meta.meta['package']['version']
+    pkg_build = meta.meta['build'].get('string', '*')
+    return pkg_name, pkg_version, pkg_build
+
+
+def get_recipe_versions():
+    """Iterate through all conda/**/meta.yaml files and return package version."""
+    conda_folder = join(MOOSE_DIR, "conda")
+    recipes = glob(join(conda_folder, "**/meta.yaml"), recursive=True)
+    for recipe in recipes:
+        yield parse_recipe_meta_data(recipe)
+
+
+def check_remote_conda():
+    """Search conda remote for all moose-libmesh package information."""
+    search_tuple = run_command(Commands.SEARCH, ["-f", "moose-libmesh", "--json"])
+    if search_tuple[2] > 0:
+        sys.exit(0)
+    else:
+        return latest_remote_libmesh_version(search_tuple[0])
+
+
+def latest_remote_libmesh_version(output):
+    """Extract version number from latest conda libmesh."""
+    results = json.loads(output)
+    latest_libmesh = results['moose-libmesh'][-1]
+    return latest_libmesh['version']
+
+
+def main():
+    """Check to see if user is using conda and environment is current."""
+
+    if not exists(CONDA_PREFIX):
+        return 0
+
+    for meta_tuple in get_recipe_versions():
+        full_name = '-'.join(meta_tuple) + '.json'
+        current_conda = join(CONDA_PREFIX, "conda-meta", full_name)
+
+        # Just handle libmesh for now. Then if needed, add
+        # functionality to handle packages with hashes in its' version.
+        if '*' in current_conda or 'None' in current_conda:
+            continue
+
+        if 'moose-libmesh' not in current_conda:
+            continue
+
+        if not exists(current_conda):
+            if check_remote_conda() is not meta_tuple[1]:
+                return 1
```

Comment:
Well, at some point we need to tell the user what's wrong :-D

Reply:
Okay let me know what your thoughts are on the new message.